lib/jarvis/thread_logger.rb

```diff
@@ -0,0 +1,42 @@
+require "cabin"
+
+module Jarvis module ThreadLogger
+  class State
+    def initialize
+      @state = {}
+      @lock = Mutex.new
+    end
+
+    def <<(hash)
+      @lock.synchronize do
+        @state[Thread.current] = hash
```

Comment:
+1 Oh I see, I think there is a default timeout to idle thread 1min (or 3mins) ?